src/tests.rs

```diff
@@ -42,18 +50,103 @@ macro_rules! assert_type {
 }
 
 #[test]
-fn primitives() {
+fn primitive_types() {
     assert_type!(bool, TypeDefPrimitive::Bool);
     assert_type!(&str, TypeDefPrimitive::Str);
     assert_type!(i8, TypeDefPrimitive::I8);
+    assert_type!(i16, TypeDefPrimitive::I16);
+    assert_type!(i32, TypeDefPrimitive::I32);
+    assert_type!(i64, TypeDefPrimitive::I64);
+    assert_type!(i128, TypeDefPrimitive::I128);
+    assert_type!(u8, TypeDefPrimitive::U8);
+    assert_type!(u16, TypeDefPrimitive::U16);
+    assert_type!(u32, TypeDefPrimitive::U32);
+    assert_type!(u64, TypeDefPrimitive::U64);
+    assert_type!(u128, TypeDefPrimitive::U128);
 
-    assert_type!([bool], TypeDefSequence::new(meta_type::<bool>()));
+    assert_type!(
+        NonZeroI8,
```

Comment:
Nitpick: the non zero types are not primitives. I would put them in their own test.

Reply:
Fixed.